pkg/client/fake/client.go

```diff
@@ -838,6 +885,81 @@ func (c *fakeClient) patch(obj client.Object, patch client.Patch, opts ...client
 	return err
 }
 
+// Applying a patch results in a deletionTimestamp that is truncated to the nearest second.
+// Check that the diff between a new and old deletion timestamp is within a reasonable threshold
+// to be considered unchanged.
+func validateTimestampDifference(newObj metav1.Object, obj metav1.Object) bool {
+	newTime := newObj.GetDeletionTimestamp()
+	oldTime := obj.GetDeletionTimestamp()
+
+	if newTime == nil || oldTime == nil {
+		return newTime == oldTime
+	}
+	return newTime.Time.Sub(oldTime.Time).Abs() < time.Second
```

Comment:
Why are we allowing any difference at all?

Reply:
Patch seems to truncate the timestamps at the second mark, so an `Equal()` comparison fails:
`s: "Rejected patch, metadata.deletionTimestamp immutable. Diff: new: 2023-05-09 11:33:21 -0400 EDT old: 2023-05-09 11:33:21.749195866 -0400 EDT m=+0.479131449",`

This function was meant to add some tolerance, but it's far from a clean solution

Reply:
Ah sorry, you wrote it in the godoc. Sounds fine!